src/bindgen/library.rs

```diff
@@ -27,6 +29,7 @@ pub struct Library {
     typedefs: ItemMap<Typedef>,
     functions: Vec<Function>,
     source_files: Vec<PathBuf>,
+    assoc_types: HashMap<AssocTypeId, (Type, u8)>,
```

Comment:
```suggestion
    assoc_types: HashMap<AssocTypeId, Type>,
```

There is no need to keep this number after parsing is finished

Reply:
Oh does this mean assoc_types will only hold fully resolved (key, conc_type) pairs? I couldn't think of anywhere else to keep the count.

Reply:
> Oh does this mean assoc_types will only hold fully resolved (key, conc_type) pairs? I couldn't think of anywhere else to keep the count.

doesn't it hold only resolved types? You shouldn't keep the count in `library.rs`. When you are constructing library with `Library::new` make sure that `result.assoc_types` is filtered